### PR TITLE
Feature/add originated syscall to all events

### PR DESCRIPTION
**Pull request by AlonZivony**
This PR needs https://github.com/aquasecurity/tracee/pull/2575/ merged first.

## Description (git log)

Add to all eBPF events the originated syscall for the event as a
member of the context.
This will allow to always k now what triggered each internal event, and
may allow for syscall events ID to not be connected to syscalls IDs.

This PR is rebased on #2575

Fixes: #2392
Close: #2389

### CODE REVIEW

**Comment by AlonZivony** [comment]
This won't pass the e2e test as it breaks the current events API

File: types/trace/trace.go
Comment (NDStrahilevitz) [inline types/trace/trace.go:43]: I'm conflicted about moving this here but if we go with this then:
1. Please fix the syscall argument filtering and move it to the context filter
2. style: I think this fits slightly better above `ReturnValue` and that just `Syscall` is enough.
3. Previously when `parse-arguments` was used, the syscall would be translated to it's string form, making it readable. This is no longer the case, and will not be present in the table printer.
4. To solve the above pointer, and keep syscalls normalized, I think we should change this to `string` and process it to a syscall name in `processEvent`.

**Review by NDStrahilevitz — CHANGES_REQUESTED** [review]

File: types/trace/trace.go
Comment (rafaeldtinoco) [reply types/trace/trace.go:43]: 1. I prefer something else than "OriginatedSyscall"
2. I agree that this should be normalized to a string.

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
Per last comment.

**Comment by AlonZivony** [comment]
I put this as a draft for now as I haven't finished all the changes requestd.

File: pkg/bufferdecoder/decoder.go
Comment (rafaeldtinoco) [inline pkg/bufferdecoder/decoder.go:64]: This conflicts with @geyslan multiscope work (both are touching the padding). @yanivagman FYIO. Since Geyslan work has priority, I believe you will have to change the approach here.

**Comment by rafaeldtinoco** [comment]
@yanivagman may we move this to next milestone ?

File: pkg/ebpf/events_pipeline.go
Comment (yanivagman) [inline pkg/ebpf/events_pipeline.go:237]: No need to call `parseContextFlags` again

File: pkg/events/events.go
Comment (yanivagman) [inline pkg/events/events.go:158]: Should handle the case of undefined ID 32bit

File: pkg/events/events.go
Comment (yanivagman) [inline pkg/events/events.go:4981]: let's keep the syscall argument (the only argument) for sys_enter/exit

**Review by yanivagman — CHANGES_REQUESTED** [review]

**Review by yanivagman — APPROVED** [review]
LGTM.
@NDStrahilevitz we can merge after your approval

File: pkg/ebpf/c/tracee.bpf.c
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:755]: Triggered the event

File: pkg/events/parse_args.go
Comment (yanivagman) [inline pkg/events/parse_args.go:79]: Why removing this?

Comment (AlonZivony) [reply pkg/events/parse_args.go:79]: Moved to the case of `open` and such

File: pkg/ebpf/c/tracee.bpf.c
Comment (NDStrahilevitz) [reply pkg/ebpf/c/tracee.bpf.c:755]: Still missing the end there
```suggestion
    u32 syscall; // The syscall which triggered the event
```

File: pkg/ebpf/events_pipeline.go
Comment (NDStrahilevitz) [inline pkg/ebpf/events_pipeline.go:338]: I don't see a case here for event which will not have an originated syscall (for example uprobe events I believe would fit this category), and that it is fine for them not to have such an origin. In that case the return should be `"", nil"`.
I assume when the syscall id passed here is 0 that would be the case, but we have to make sure the bpf code sets it as such.

File: pkg/events/parse_args.go
Comment (NDStrahilevitz) [inline pkg/events/parse_args.go:59]: Just a question:
So these events will have `context.Syscall="enter"/"exit"` and their argument will be the actual called syscall right?

File: pkg/filters/context.go
Comment (NDStrahilevitz) [inline pkg/filters/context.go:173]: This should probably be higher in the chain so it will get short circuited faster.

File: pkg/events/parse_args.go
Comment (AlonZivony) [reply pkg/events/parse_args.go:59]: No, the argument and the context value will be matching.
We kept it as is because we want them to be printed in the Table Printer.

Comment (NDStrahilevitz) [reply pkg/events/parse_args.go:59]: Wouldn't it be better to add a Syscall column instead? Actually, now all events with syscall arguments will be missing the syscall in the table printer.

Comment (AlonZivony) [reply pkg/events/parse_args.go:59]: You are correct, but the `syscall` argument is used in more complicated scenarios so in those cases I think using `json` or `gob` is acceptable.
I think we didn't want to make the Table View more packed as it has too many values to print as it is.

**Comment by NDStrahilevitz** [comment]
@AlonZivony Needs a rebase.

**Review by NDStrahilevitz — APPROVED** [review]
LGTM, I will merge once tests are done

**Pull request merged**
